### PR TITLE
ORC-867: Upgrade hive-storage-api to 2.8.1

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to upgrade `hive-storage-api` from 2.7.2 to 2.8.1.

### Why are the changes needed?

This will bring the latest improvement and bug fixes.

### How was this patch tested?

Pass the CIs.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @williamhyun and @pavibhai

**Review by kbendick — APPROVED** [review]
This is great. Thank you @dongjoon-hyun!

Is there anything we can change / remove from the build (pom) file? I took a look, and it didn't seem like it as the only explicit dependency is on the commons-lang3, but felt worth bringing up just in case.

**Comment by dongjoon-hyun** [comment]
Thank you for review and approval, @kbendick . 
Since `commons-lang3` is already in `core` module, there is nothing to change for this.
- https://github.com/apache/orc/blob/main/java/core/pom.xml#L48

**Comment by dongjoon-hyun** [comment]
Oh..  `guava` dependency seems to be promoted from `test` dependency to `compile` dependency.
Let me check my PR to `Draft` first.

File: java/core/src/test/org/apache/orc/impl/TestRecordReaderImpl.java [hunk 1]
```diff
@@ -310,6 +310,9 @@ public Object getLiteral() {
       return null;
     }
 
+    @Override
```
Comment (pavibhai) [inline java/core/src/test/org/apache/orc/impl/TestRecordReaderImpl.java:313]: Looks like we will do another release of StorageAPI. I was wondering if we should push this as a default?

Comment (dongjoon-hyun) [reply java/core/src/test/org/apache/orc/impl/TestRecordReaderImpl.java:313]: Yes, we need 2.8.1. This PR will stay here and will retarget to 2.8.1, @pavibhai .

BTW, what do you mean by the following?
> we should push this as a default?

Comment (pavibhai) [reply java/core/src/test/org/apache/orc/impl/TestRecordReaderImpl.java:313]: I was wondering if we should add a default implementation to this new method into PredicateLeaf. But not required if this is the only failure we are seeing.

**Comment by dongjoon-hyun** [comment]
For this one, we will retry when `hive-storage-api` 2.8.1 is ready.
-

**Review by omalley — APPROVED** [review]
LGTM, other than the pom needs to switch to storage-api 2.8.1. Build & unit tests run fine.

**Comment by dongjoon-hyun** [comment]
Thank you, @omalley . I also saw that 2.8.1 is published already.
- https://repo1.maven.org/maven2/org/apache/hive/hive-storage-api/2.8.1/

The pom is updated and I rebased this PR.

**Comment by dongjoon-hyun** [comment]
If CIs pass, I'll merge this to main/branch-1.7.

**Comment by dongjoon-hyun** [comment]
All Java test passed with GitHub Action (Java8/11/16/17-ea).

**Pull request merged**